yang/vyatta-protocols-frr-isis-v1.yang

```diff
@@ -0,0 +1,584 @@
+      leaf level-1 {
+        type uint16 {
+          range "1..120";
+        }
+        units "seconds";
+        configd:help "Minimum time between consecutive level-1 SPFs.";
+        description "Minimum time between consecutive level-1 SPFs.";
+      }
+      leaf level-2 {
+        type uint16 {
+          range "1..120";
+        }
+        units "seconds";
+        configd:help "Minimum time between consecutive level-2 SPFs.";
+        description "Minimum time between consecutive level-2 SPFs.";
+      }
+    }
+
+    leaf is-type {
+      type level;
+      configd:help "IS Level for this routing process (OSI only)";
+    }
+
+    leaf metric-style {
+      type metric-style-type;
+      description "Define the style of TLVs metric supported.";
+      configd:help "Define the style of TLVs metric supported.";
+    }
+
+    leaf lsp-mtu {
+      type uint16 {
+        range "128..4352";
+      }
+      configd:help "MTU of an LSP.";
+      description "MTU of an LSP.";
+    }
+  }
+
+  grouping redistribute-protocol {
+    leaf kernel {
+      type level;
+      configd:help "Kernel routes (not installed via the zebra RIB).";
+      description "Kernel routes (not installed via the zebra RIB).";
+    }
+    leaf connected {
+      type level;
+      configd:help "Connected routes (directly attached subnet or host).";
+      description "Connected routes (directly attached subnet or host).";
+    }
+    leaf static {
+      type level;
+      configd:help "Statically configured routes.";
+      description "Statically configured routes.";
+    }
+    leaf ospf {
+      type level;
+      configd:help "Open Shortest Path First (OSPFv2).";
+      description "Open Shortest Path First (OSPFv2).";
+    }
+    leaf bgp {
+      type level;
+      configd:help "Redistribute bgp routes";
+      description "Redistribute bgp routes";
+    }
+    leaf rip {
+      type level;
+      configd:help "Redistribute rip routes";
+      description "Redistribute rip routes";
+    }
+  }
+
+  grouping isis-redistribute {
+    container redistribute {
+      configd:help "Redistributes routes learned from other routing protocols.";
+      description "Redistributes routes learned from other routing protocols.";
+      container ipv4 {
+        configd:help "IPv4 route redistribution.";
+        description "IPv4 route redistribution.";
+        uses redistribute-protocol;
+      }
+      container ipv6 {
+        configd:help "IPv6 route redistribution.";
+        description "IPv6 route redistribution.";
+        uses redistribute-protocol;
+      }
+    }
+  }
+
+  augment /protocols:protocols {
+    list isis {
+      configd:help "Intermediate System to Intermediate System (ISIS) parameters";
+      description "Intermediate System to Intermediate System (ISIS) parameters";
+      must "(as-num != 23456)" {
+        error-message "Invalid AS Number, 23456 (Reserved)";
+      }
+      must "count(../isis/as-num) <= 1" {
```

Comment:
Better expressed as 'max-elements 1'

Reply:
We found that ISIS allows multiple instances so we have removed the "must" statement as it is not required. 

Thanks,

Reply:
We found that ISIS allows multiple instances so we have removed the "must" statement as it is not required. 

Thanks, 


> Approved YANG files as part of data model review process.

Thanks for the approval.